### PR TITLE
Update ncurses to not build a separate libtinfo but provide a soft link instead. Force linking to ncurses in libreadline

### PR DESCRIPTION
**Pull request by ocaisa**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by ocaisa** [comment]
Test report by @ocaisa
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
jrl01 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/7951fbf08ea2e6dc1deee3767ba82e0e for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
alanc-VirtualBox - Linux ubuntu 16.04, Intel(R) Core(TM) i7-4500U CPU @ 1.80GHz, Python 2.7.12
See https://gist.github.com/1efb53a376ca10dfe9afcbadc9b7b14a for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
alanc-VirtualBox - Linux ubuntu 16.04, Intel(R) Core(TM) i7-4500U CPU @ 1.80GHz, Python 2.7.12
See https://gist.github.com/968edbace1fa8563583ac9559cc0b874 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
alanc-VirtualBox - Linux ubuntu 16.04, Intel(R) Core(TM) i7-4500U CPU @ 1.80GHz, Python 2.7.12
See https://gist.github.com/300e594b5986d81d38fc53f21a77548a for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**FAILED**
Build succeeded for 2 out of 3 (3 easyconfigs in this PR)
alanc-VirtualBox - Linux ubuntu 16.04, Intel(R) Core(TM) i7-4500U CPU @ 1.80GHz, Python 2.7.12
See https://gist.github.com/a9cac44bbe3246269d836f6a9a280b98 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**FAILED**
Build succeeded for 3 out of 4 (3 easyconfigs in this PR)
jrl01 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/de91d6f0f2e4d9872b94194817b61510 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
alanc-VirtualBox - Linux ubuntu 16.04, Intel(R) Core(TM) i7-4500U CPU @ 1.80GHz, Python 2.7.12
See https://gist.github.com/8c57b9471e6450c0020e7bf7c9c52067 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in this PR)
jrl01 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/a73a1ae3cb17a2a5e5de426b8d7c11a0 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
alanc-VirtualBox - Linux ubuntu 16.04, Intel(R) Core(TM) i7-4500U CPU @ 1.80GHz, Python 2.7.12
See https://gist.github.com/71b08e832d573ed11022aa39d8b48989 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in this PR)
jrl01 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/779ab5a1d8c5f75d06185fdf8767151c for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
jrl01 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/4fba127129f573890d9d83c059e888da for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
jrl01 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/3b403327fbdf397c5666f279f5765e55 for a full test report.

**Comment by ocaisa** [comment]
Test report by @ocaisa
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
alanc-VirtualBox - Linux ubuntu 16.04, Intel(R) Core(TM) i7-4500U CPU @ 1.80GHz, Python 2.7.12
See https://gist.github.com/453de02d373fc10d96b0693bcbf3a6c1 for a full test report.

File: easybuild/easyconfigs/n/ncurses/ncurses-6.0.eb [hunk 1]
```diff
@@ -18,13 +17,22 @@ patches = ['ncurses-%(version)s_gcc-5.patch']
...
     # the UTF-8 enabled version (ncursesw)
-    '--with-shared --enable-overwrite --with-termlib ' +
-    '--enable-ext-colors --enable-widec --includedir=%(installdir)s/include/ncursesw/'
+    '--with-shared --enable-overwrite --enable-ext-colors --enable-widec --includedir=%(installdir)s/include/ncursesw/'
 ]
 
-libs = ["form", "menu", "ncurses", "panel", "tinfo"]
+# need to take care of $CFLAGS ourselves with dummy toolchain
+# we need to add -fPIC, but should also include -O* option to avoid compiling with -O0 (default for GCC)
+buildopts = ['CFLAGS="-O2 -fPIC"', 'CFLAGS="-O2 -fPIC"']
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-6.0.eb]: you don't need to make this a list, just a string will work (it will be used once for every entry in `configopts`)

```python
buildopts = 'CFLAGS="-O2 -fPIC"
```

File: easybuild/easyconfigs/n/ncurses/ncurses-6.0.eb [hunk 2]
```diff
@@ -18,13 +17,22 @@ patches = ['ncurses-%(version)s_gcc-5.patch']
...
+
+# Symlink to ncurses for libtinfo (since it can handle the API) so it doesn't get picked up from the OS
```
Comment (boegel) [inline easybuild/easyconfigs/n/ncurses/ncurses-6.0.eb]: I'd rephrase this to `symlink libtinfo to libncurses (since ...`?

**Comment by boegel** [comment]
Other than the minor remarks, this looks good to me. I don't consider symlinking `libtinfo` to `libncurses` a dirty hack at all...

You mentioned in https://github.com/easybuilders/easybuild-easyconfigs/pull/3545#r135517723 that Debian "has done this in the past". Are you saying they are not doing this anymore, and if so, why not, and what are they doing now? Any references?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
node2497.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/c0b51919108f3a01891ecb7d33625b5c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 12 out of 12 (12 easyconfigs in this PR)
node2125.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/f4d24b4e769911130073e2e01b931246 for a full test report.

**Comment by ocaisa** [comment]
I was only reading that from https://bugzilla.redhat.com/show_bug.cgi?id=499837#c2 (which was in 2010), I don't know what the current status is.

**Comment by boegel** [comment]
Seems like they provide a separate `libtinfo.so` now (not a symlink to `libncurses.so`, since they're split across different pacakges: `libncurses5-dev` which requires `libtinfo-dev` as a dep: https://packages.debian.org/stretch/amd64/libtinfo-dev/filelist, https://packages.debian.org/stretch/amd64/libncurses5-dev/filelist

**Comment by ocaisa** [comment]
So do you think we should be doing the same?

**Comment by boegel** [comment]
@ocaisa No, not really, if this approach works, fine. We had the split `libtinfo` before, and that caused problems in our context...

**Comment by ocaisa** [comment]
Ok, my bad, I thought you meant to give `ncurses` a `termcap` dep (but `termcap` hasn't been updated since 2002)

File: easybuild/easyconfigs/l/libreadline/libreadline-6.3.eb [hunk 3]
```diff
@@ -19,8 +18,11 @@ patches = ['libreadline-%(version)s-bugfix.patch']
 
 dependencies = [('ncurses', '6.0')]
 
+# need to take care of $CFLAGS ourselves with dummy toolchain
+# we need to add -fPIC, but should also include -O* option to avoid compiling with -O0 (default for GCC)
+buildopts = "CFLAGS='-O2 -fPIC' "
 # for the termcap symbols, use EB ncurses
-preconfigopts = "env LDFLAGS='-lncurses'"
+buildopts += "SHLIB_LIBS='-lncurses'"
```
Comment (ocaisa) [inline easybuild/easyconfigs/l/libreadline/libreadline-6.3.eb]: From http://www.linuxfromscratch.org/lfs/view/7.4/chapter06/readline.html

**Comment by boegel** [comment]
Going in, thanks @ocaisa!
